util/src/main/java/io/kubernetes/client/util/ModelMapper.java

```diff
@@ -485,14 +486,26 @@ private static List<String> getClassNamesFromPackage(ClassLoader classLoader, St
 
   private static void processJarPackage(URL packageURL, String packageName, String pkg, ArrayList<String> names) throws IOException {
     String jarFileName = URLDecoder.decode(packageURL.getFile(), "UTF-8");
-    jarFileName = jarFileName.substring(5, jarFileName.indexOf("!"));
+    JarFile jf = null;
+    // jar: client in repository; nested: client in a fat jar
+    if (jarFileName.startsWith("jar:") || jarFileName.startsWith("nested:")) {
+      jf = ((JarURLConnection) packageURL.openConnection()).getJarFile();
+    }
+    // file: client is a file in target (unit test)
+    if (jarFileName.startsWith("file:") ) {
+      jarFileName = jarFileName.substring(5, jarFileName.indexOf("!"));
+      jf = new JarFile(jarFileName);
+    }
+    if (jf == null) {
+      logger.error("Loading classes from jar with error packageURL: {}", jarFileName);
```

Comment:
```suggestion
      logger.error("Loading classes from jar with error packageURL: {}", packageURL);
```
nit: looks like we might want to dump the packageURL here?

Reply:
Thx

Reply:
the same with packageURL